static/js/libs/fileValidation.js

```diff
@@ -0,0 +1,106 @@
+function baseRestrictions(file, restrictions) {
+  return new Promise((resolve, reject) => {
+    const errors = [];
+    if (restrictions.accept) {
+      if (restrictions.accept.indexOf(file.type) === -1) {
+        errors.push("File type is incorrect");
+      }
+    }
+
+    if (restrictions.size) {
+      const fileSize = file.size / 1000000;
+      if (restrictions.size.max && fileSize > restrictions.size.max) {
+        errors.push(`File size is over ${restrictions.size.max}MB`);
+      }
+      if (restrictions.size.min && fileSize < restrictions.size.min) {
+        errors.push(`File size is below ${restrictions.size.min}MB`);
+      }
+    }
+
+    if (errors.length > 0) {
+      reject(errors);
+    } else {
+      resolve(file);
+    }
+  });
+}
+
+function imageRestrictions(file, restrictions) {
+  if (!restrictions.accept || restrictions.accept[0].indexOf("image") < 0) {
+    return file;
```

Comment:
Why returning `file`? It's quite a different return value from `Promise`, is it expected that return value is checked every time it's used?

Reply:
I was under the impression (I can't remember where from) that you can return a normal value within a promise chain. I could be wrong, but this does seem to work... Or I haven't tested it thoroughly enough

Reply:
Maybe when Promise is expected a value can be treated as resolved promise... but `imageRestrictions().then()` would not work if it returns file I guess?

Reply:
Yeah I think that's how it works. Probably better to resolve the promise for compatibility - will update